### PR TITLE
add w4mrutils R package

### PR DESCRIPTION
**Pull request by Lain-inrae**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by Lain-inrae** [comment]
Hello, @conda-forge/help-r !
Can you review my pr, please?
Thanks!

**Comment by Lain-inrae** [comment]
@conda-forge-admin, please ping team

**Comment by Lain-inrae** [comment]
@conda-forge/help-r, please ping team

**Comment by Lain-inrae** [comment]
@conda-forge/help-r, please ping conda-forge/help-r

**Comment by Lain-inrae** [comment]
@conda-forge-admin, please ping conda-forge/help-r

**Comment by Lain-inrae** [comment]
@conda-forge-admin, please ping team

**Comment by ocefpaf** [comment]
@conda-forge/r we need some R experts here :-)

File: recipes/w4mrutils/conda_build_config.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+R_version:
```
Comment (xhochy) [inline recipes/w4mrutils/conda_build_config.yaml:1]: This shouldn't be needed. We will take the R version from the global pinning.

File: recipes/w4mrutils/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+package:
+  name: r-w4mrutils
+  version: "{{ version }}"
+
+source:
+  url: https://github.com/workflow4metabolomics/W4MRUtils/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: b6c2f7b3aae2bb88a8f27db40596a76bcd5f84b2d1324f8147787b070087c037
+  git_rev: v{{ version }}
+  git_depth: -1
```
Comment (xhochy) [inline recipes/w4mrutils/meta.yaml:11]:
```suggestion
```

File: recipes/w4mrutils/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+build:
+  noarch: generic
+  merge_build_host: true
```
Comment (xhochy) [inline recipes/w4mrutils/meta.yaml:15]:
```suggestion
```

File: recipes/w4mrutils/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  number: 0
+  script: R CMD INSTALL --build .
```
Comment (xhochy) [inline recipes/w4mrutils/meta.yaml:17]: Please use the same build scripts as other packages use.

File: recipes/w4mrutils/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+requirements:
+  build:
+    - r-base {{ R_version }}
+  run:
+    - r-base {{ R_version }}
```
Comment (xhochy) [inline recipes/w4mrutils/meta.yaml:23]:
```suggestion
  host:
    - r-base
  run:
    - r-base
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by Lain-inrae** [comment]
@conda-forge-admin, please ping team

**Comment by Lain-inrae** [comment]
@conda-forge-admin, please ping team

Please, someone can review my PR please?

**Comment by mfansler** [comment]
We prefer to work downstream of CRAN. Is there a reason why this can't first go to CRAN?

**Comment by Lain-inrae** [comment]
This PR complies to everything on the checklist.
Is there a reasonable justification to refuse to publish my package, after five month of wait?

My work on this package is on standby because of the delay.
This package was meant to be a toolset to help some people in the organization I work at, to create processing tools before the end of my contract. Five months is a very long delay regarding the duration of my contract.

I will change the way I download the package in the next release, but please, can you pull my request?

**Comment by mfansler** [comment]
> This PR complies to everything on the checklist. Is there a reasonable justification to refuse to publish my package, after five month of wait?

Five months is exceptionally long for this to sit in waiting - I'm sorry about that. Please understand that we are all volunteers here and the R team is a small (and IRL very busy) subteam that struggles with time allocation (at least I do!). 

R packages that come from CRAN make it simple for us to process them and have an implied upstream vetting of the quality of the package. When a package doesn't have that, it means more time commitment because it is expected that both the recipe *and* the package need to be evaluated. Moreover, we don't have formalized rules for evaluating a *package*, which in turn means more effort. Personally, if I see in my queue a mature package with thousands of downloads that is having a failure versus a staged-recipes submission with an unclear time commitment to review, I'm going to work the former.

Generally, Conda Forge packages are [meant](https://conda-forge.org/docs/orga/guidelines.html#intended-usage) to be releases ready for end-users. The early-lifecycle status of the package (v0.0.1) raises concern that it is not something meant to be actively used, but is just a placeholder for a *developing* package. This doesn't entail rejection, but I want to emphasize that development versions (e.g., "dailies") should not be released to Conda Forge. This is another reason why we prefer working from CRAN, because it creates some additional hurdle to be overcome before our CI resources are engaged.

Another review concern I have is whether this package is eventually going to depend on a Bioconductor package given that it seems bioinfo-related. If so, it should go to Bioconda instead. Migrating Conda Forge -> Bioconda is problematic because users prioritize Conda Forge in their configurations.

Please let me know if these are applicable concerns or not. Otherwise, the package appears *technically* acceptable.

> My work on this package is on standby because of the delay. This package was meant to be a toolset to help some people in the organization I work at, to create processing tools before the end of my contract. Five months is a very long delay regarding the duration of my contract.

It is unclear why a *Conda Forge feedstock* is necessary for this. Anaconda Cloud provides both free personal and organization accounts, which is how I distribute my own early-lifecycle Conda packages and frequently how smaller scientific groups distribute Conda versions of their software (e.g., [Fermi](https://anaconda.org/fermi), [psi4](https://anaconda.org/psi4)). Alternatively, since you appear to be working in bioinformatics space, [the Bioconda repository](https://bioconda.github.io) would be an appropriate platform for distribution and is far more lenient about package maturity.  It is not uncommon for packages to transition from Bioconda to Conda Forge when they shift to be of broader interest.

> I will change the way I download the package in the next release, but please, can you pull my request?

I'm fine with merging under the assurance that it will eventually shift to a CRAN upstream.

**Comment by Lain-inrae** [comment]
> Five months is exceptionally long for this to sit in waiting - I'm sorry about that. Please understand that we are all volunteers here and the R team is a small (and IRL very busy) subteam that struggles with time allocation (at least I do!).

I don't blame people for not having time to allocate to conda-forge. I understand why it is complicated.
It is just really frustrating to have to wait for a PR to be accepted, and after months of wait, to have a message telling to change the upstream. And you will better understand why with the rest of the message.

> Generally, Conda Forge packages are meant to be releases ready for end-users. The early-lifecycle status of the package (v0.0.1) raises concern that it is not something meant to be actively used

This is the kind of package that - by essence - cannot be used before it is on conda. People (at Workflow4Metabolomics) will not create tools based on a package that cannot be distributed, because it would break their tools.

Furthermore, this package is meant to be a standard dependency for every tool we use. Each and every one of them.
And it is part of my work to make sure it is.

 > Another review concern I have is whether this package is eventually going to depend on a Bioconductor package given that it seems bioinfo-related.
 
 That's both true and untrue. It is bio-related because it will be used by bio-informaticians to creates processing tools. But is is not because it will not contain any bioinfo-related processing.
I'm not a bio-informatician. I am a developper whose work is to uniformize the way tools work in our organisation, and help people create better code/tools. This package is meant to provide a set of utilities to help them create better tools. It is not specific to bio-info. 
 
 > It is unclear why a Conda Forge feedstock is necessary for this. Anaconda Cloud provides both free personal and organization accounts, which is how I distribute my own early-lifecycle Conda packages and frequently how smaller scientific groups distribute Conda versions of their software
 
 I cannot use channels other than bioconda or conda-forge because our software are installed on usegalaxy.fr (and usergalaxy.eu) and I cannot use an other upstream because galaxy uses only conda to automatically deploy tool's dependencies/requirements. Those galaxy instances does not allow to use custom channels, and does not allow to manually install tools neither. So, the only ways for us to install tools is to have their requirements on conda. Thus the delay in my work. If I had an other solution, I would have used it.
 
 > I'm fine with merging under the assurance that it will eventually shift to a CRAN upstream.
 It will at the next version.

**Review by mfansler — APPROVED** [review]

**Comment by fgiacomoni** [comment]
Thanks Marvin

**Comment by Lain-inrae** [comment]
Thank you, mfansler!

**Pull request merged**